### DISCUSSION TITLE
How to handle accessing states that don't exist yet

### DISCUSSION THREAD

**Discussion by sebsch1337** [neuefische/questions#276] [Web] [answered]
Hey!

I'm using a useLocalStorage hook that returns a state. Unfortunately the components where i use the states try to access the state when it doesnt exist yet. When the state is created from the localStorage (after a few re-renderings) the components get the proper state.
But in the meantime the app crashed already, as the components try to access the non existing states.

How can i handle accessing a state in the meantime, when it doesnt exist yet?

```
export default function Details() {  
  const [destination, setDestination] = useLocalStorage("destination", dummyDestination);
  
  return <title>{destination}</title>;
}
```

Thanks in advance!

**Comment by mbosselmann** [comment]
Hi Sebastian,
can you help us with providing a PR to your current state in your capstone project? That would be really helpful to get a better understanding of your problem.

**Comment by sebsch1337** [comment]
https://github.com/sebsch1337/capstone-project/pull/15

file: pages/details/[id].js

Navigate to THAILAND > BANGKOK
=> In this case it will work

When you refresh the page or visit http://localhost:3000/details/br43kd8i again
=> It will crash

**Comment by mbosselmann** [comment, marked as answer]
Sebastian and I worked on a solution that can handle an undefined value of the const `destination` during the first rendering of the details page. We used an `if statement` to ensure that the `constants` and the `JSX` are only rendered if the const `destination` is truthy:

```js
if (destination) {
    const startDate = new Date(destination.startDate * 1000).toISOString().substring(0, 10);
    const endDate = new Date(destination.endDate * 1000).toISOString().substring(0, 10);

    return (
       // JSX
);
}
```
